### ISSUE TITLE
CH:EMED validation problem with extensible binding?

### ISSUE THREAD

**Issue by oliveregger** [ahdis/matchbox#250]
https://fhir.ch/ig/ch-emed/Bundle-PharmaceuticalAdvice-ChangeDosage.json.html

returns an error because Composition.type.coding has a loinc code defined with "61356-2", binding is extensible
this should be however allowed and did not give an error at the time of publication:

https://fhir.ch/ig/ch-emed/StructureDefinition-ch-emed-composition-pharmaceuticaladvice.html

need to check what has changed:

```json
{
    "resourceType": "OperationOutcome",
    "id": "OperationOutcome/1880cc46fafbb27ac8226523fe8c89b47668224d099f503b4712a38298b2a0bc",
    "issue": [
        {
            "extension": [
                {
                    "extension": [
                        {
                            "url": "profile",
                            "valueUri": "http://fhir.ch/ig/ch-emed/StructureDefinition/ch-emed-document-pharmaceuticaladvice"
                        },
                        {
                            "url": "profileVersion",
                            "valueUri": "5.0.0-ballot"
                        },
                        {
                            "url": "profileDate",
                            "valueDateTime": "2024-05-16T19:08:44+00:00"
                        },
                        {
                            "url": "total",
                            "valueDuration": {
                                "value": 25271,
                                "unit": "ms"
                            }
                        },
                        {
                            "url": "validatorVersion",
                            "valueString": "powered by matchbox 3.8.5, hapi-fhir 7.0.2 and org.hl7.fhir.core 6.3.5"
                        },
                        {
                            "url": "ig",
                            "valueString": "ch.fhir.ig.ch-emed#5.0.0-ballot"
                        },
                        {
                            "url": "hintAboutNonMustSupport",
                            "valueBoolean": false
                        },
                        {
                            "url": "recursive",
                            "valueBoolean": false
                        },
                        {
                            "url": "showMessagesFromReferences",
                            "valueBoolean": false
                        },
                        {
                            "url": "doDebug",
                            "valueBoolean": false
                        },
                        {
                            "url": "assumeValidRestReferences",
                            "valueBoolean": false
                        },
                        {
                            "url": "canDoNative",
                            "valueBoolean": false
                        },
                        {
                            "url": "noExtensibleBindingMessages",
                            "valueBoolean": false
                        },
                        {
                            "url": "noUnicodeBiDiControlChars",
                            "valueBoolean": false
                        },
                        {
                            "url": "noInvariants",
                            "valueBoolean": false
                        },
                        {
                            "url": "displayIssuesAreWarnings",
                            "valueBoolean": true
                        },
                        {
                            "url": "wantInvariantsInMessages",
                            "valueBoolean": false
                        },
                        {
                            "url": "doImplicitFHIRPathStringConversion",
                            "valueBoolean": false
                        },
                        {
                            "url": "securityChecks",
                            "valueBoolean": false
                        },
                        {
                            "url": "crumbTrails",
                            "valueBoolean": false
                        },
                        {
                            "url": "forPublication",
                            "valueBoolean": false
                        },
                        {
                            "url": "httpReadOnly",
                            "valueBoolean": false
                        },
                        {
                            "url": "allowExampleUrls",
                            "valueBoolean": false
                        },
                        {
                            "url": "txServer",
                            "valueUri": "http://tx.fhir.org"
                        },
                        {
                            "url": "fhirVersion",
                            "valueString": "4.0.1"
                        },
                        {
                            "url": "ig",
                            "valueString": "ch.fhir.ig.ch-emed#5.0.0-ballot"
                        },
                        {
                            "url": "locale",
                            "valueString": "English"
                        },
                        {
                            "url": "jurisdiction",
                            "valueString": "urn:iso:std:iso:3166#CH"
                        },
                        {
                            "url": "extensions",
                            "valueString": "any"
                        },
                        {
                            "url": "sessionId",
                            "valueString": "-1653045309"
                        },
                        {
                            "url": "package",
                            "valueString": "hl7.fhir.xver-extensions#4.0"
                        },
                        {
                            "url": "package",
                            "valueString": "hl7.fhir.r4.core#4.0.1"
                        },
                        {
                            "url": "package",
                            "valueString": "hl7.fhir.uv.extensions.r4#1.0.0"
                        },
                        {
                            "url": "package",
                            "valueString": "hl7.terminology#5.4.0"
                        },
                        {
                            "url": "package",
                            "valueString": "ihe.formatcode.fhir#1.2.0"
                        },
                        {
                            "url": "package",
                            "valueString": "ch.fhir.ig.ch-term#3.0.0"
                        },
                        {
                            "url": "package",
                            "valueString": "ch.fhir.ig.ch-core#5.0.0-ballot"
                        },
                        {
                            "url": "package",
                            "valueString": "ch.fhir.ig.ch-emed#5.0.0-ballot"
                        }
                    ],
                    "url": "http://matchbox.health/validation"
                }
            ],
            "severity": "information",
            "code": "informational",
            "diagnostics": "Validation for profile http://fhir.ch/ig/ch-emed/StructureDefinition/ch-emed-document-pharmaceuticaladvice|5.0.0-ballot (2024-05-16T19:08:44+00:00). Loaded packages: hl7.fhir.xver-extensions#4.0, hl7.fhir.r4.core#4.0.1, hl7.fhir.uv.extensions.r4#1.0.0, hl7.terminology#5.4.0, ihe.formatcode.fhir#1.2.0, ch.fhir.ig.ch-term#3.0.0, ch.fhir.ig.ch-core#5.0.0-ballot, ch.fhir.ig.ch-emed#5.0.0-ballot. Duration: 25.271s. powered by matchbox 3.8.5, hapi-fhir 7.0.2 and org.hl7.fhir.core 6.3.5. Validation parameters: CliContext{doNative=false, extensions=[any], hintAboutNonMustSupport=false, recursive=false, showMessagesFromReferences=false, doDebug=false, assumeValidRestReferences=false, canDoNative=false, noExtensibleBindingMessages=false, noUnicodeBiDiControlChars=false, noInvariants=false, displayIssuesAreWarnings=true, wantInvariantsInMessages=false, doImplicitFHIRPathStringConversion=false, htmlInMarkdownCheck=WARNING, txServer='http://tx.fhir.org', lang='null', snomedCT='null', fhirVersion='4.0.1', ig='ch.fhir.ig.ch-emed#5.0.0-ballot', questionnaireMode=CHECK, level=HINTS, mode=VALIDATION, securityChecks=false, crumbTrails=false, forPublication=false, allowExampleUrls=false, locale='English', locations={}, jurisdiction='urn:iso:std:iso:3166#CH', igsPreloaded=null, onlyOneEngine=false, httpReadOnly=false}"
        },
        {
            "extension": [
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-line",
                    "valueInteger": 1
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-col",
                    "valueInteger": 2585
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-source",
                    "valueString": "InstanceValidator"
                }
            ],
            "severity": "information",
            "code": "informational",
            "diagnostics": "This element does not match any known slice defined in the profile http://fhir.ch/ig/ch-emed/StructureDefinition/ch-emed-document-pharmaceuticaladvice|5.0.0-ballot (this may not be a problem, but you should check that it's not intended to match a slice) Slice info: 1.) Bundle.entry[0]: Does not match slice 'Composition' (discriminator: resource.conformsTo('http://fhir.ch/ig/ch-emed/StructureDefinition/ch-emed-composition-pharmaceuticaladvice')) 2.) Bundle.entry[0]: Bundle.entry[0].resource.type: Der angegebene Code 'http://loinc.org#61356-2 ('Medication pharmaceutical advice.extended Document')' befindet sich nicht im ValueSet 'http://fhir.ch/ig/ch-term/ValueSet/DocumentEntry.typeCode|3.0.0'; Der Displayname f\\u00fcr http://loinc.org#61356-2 sollte einer von ''Medication pharmaceutical advice.extended Document' (en-US)' anstelle von 'Medication pharmaceutical advice.extended Document' sein. 3.) Bundle.entry[0]: Does not match slice 'Patient' (discriminator: resource.conformsTo('http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-patient')) 4.) Bundle.entry[0]: Bundle.entry[0].resource: Specified profile type was 'Patient' in profile 'http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-patient|5.0.0-ballot', but found type 'Composition' 5.) Bundle.entry[0]: Does not match slice 'Practitioner' (discriminator: resource.conformsTo('http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-practitioner')) 6.) Bundle.entry[0]: Bundle.entry[0].resource: Specified profile type was 'Practitioner' in profile 'http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-practitioner|5.0.0-ballot', but found type 'Composition' 7.) Bundle.entry[0]: Does not match slice 'PractitionerRole' (discriminator: resource.conformsTo('http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-practitionerrole')) 8.) Bundle.entry[0]: Bundle.entry[0].resource: Specified profile type was 'PractitionerRole' in profile 'http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-practitionerrole|5.0.0-ballot', but found type 'Composition' 9.) Bundle.entry[0]: Does not match slice 'Organization' (discriminator: resource.conformsTo('http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-organization')) 10.) Bundle.entry[0]: Bundle.entry[0].resource: Specified profile type was 'Organization' in profile 'http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-organization|5.0.0-ballot', but found type 'Composition' 11.) Bundle.entry[0]: Does not match slice 'RelatedPerson' (discriminator: resource.conformsTo('http://hl7.org/fhir/StructureDefinition/RelatedPerson')) 12.) Bundle.entry[0]: Bundle.entry[0].resource: Specified profile type was 'RelatedPerson' in profile 'http://hl7.org/fhir/StructureDefinition/RelatedPerson|4.0.1', but found type 'Composition' 13.) Bundle.entry[0]: Does not match slice 'Device' (discriminator: resource.conformsTo('http://hl7.org/fhir/StructureDefinition/Device')) 14.) Bundle.entry[0]: Bundle.entry[0].resource: Specified profile type was 'Device' in profile 'http://hl7.org/fhir/StructureDefinition/Device|4.0.1', but found type 'Composition' 15.) Bundle.entry[0]: Does not match slice 'Encounter' (discriminator: resource.conformsTo('http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-encounter')) 16.) Bundle.entry[0]: Bundle.entry[0].resource: Specified profile type was 'Encounter' in profile 'http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-encounter|5.0.0-ballot', but found type 'Composition' 17.) Bundle.entry[0]: Does not match slice 'Location' (discriminator: resource.conformsTo('http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-location')) 18.) Bundle.entry[0]: Bundle.entry[0].resource: Specified profile type was 'Location' in profile 'http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-location|5.0.0-ballot', but found type 'Composition' 19.) Bundle.entry[0]: Does not match slice 'Binary' (discriminator: resource.conformsTo('http://hl7.org/fhir/StructureDefinition/Binary')) 20.) Bundle.entry[0]: Bundle.entry[0].resource: Specified profile type was 'Binary' in profile 'http://hl7.org/fhir/StructureDefinition/Binary|4.0.1', but found type 'Composition' 21.) Bundle.entry[0]: Does not match slice 'Observation' (discriminator: resource.conformsTo('http://fhir.ch/ig/ch-emed/StructureDefinition/ch-emed-observation')) 22.) Bundle.entry[0]: Bundle.entry[0].resource: Specified profile type was 'Observation' in profile 'http://fhir.ch/ig/ch-emed/StructureDefinition/ch-emed-observation|5.0.0-ballot', but found type 'Composition' 23.) Bundle.entry[0]: Does not match slice 'MedicationStatementChanged' (discriminator: resource.conformsTo('http://fhir.ch/ig/ch-emed/StructureDefinition/ch-emed-medicationstatement-changed')) 24.) Bundle.entry[0]: Bundle.entry[0].resource: Specified profile type was 'MedicationStatement' in profile 'http://fhir.ch/ig/ch-emed/StructureDefinition/ch-emed-medicationstatement-changed|5.0.0-ballot', but found type 'Composition' 25.) Bundle.entry[0]: Does not match slice 'MedicationRequestChanged' (discriminator: resource.conformsTo('http://fhir.ch/ig/ch-emed/StructureDefinition/ch-emed-medicationrequest-changed')) 26.) Bundle.entry[0]: Bundle.entry[0].resource: Specified profile type was 'MedicationRequest' in profile 'http://fhir.ch/ig/ch-emed/StructureDefinition/ch-emed-medicationrequest-changed|5.0.0-ballot', but found type 'Composition'",
            "expression": [
                "Bundle.entry[0]"
            ]
        },
        {
            "extension": [
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-line",
                    "valueInteger": 1
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-col",
                    "valueInteger": 7771
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-source",
                    "valueString": "InstanceValidator"
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-message-id",
                    "valueCode": "Validation_VAL_Profile_Minimum_SLICE"
                }
            ],
            "severity": "error",
            "code": "structure",
            "diagnostics": "Slice 'Bundle.entry:Composition': minimum required = 1, but only found 0 (from http://fhir.ch/ig/ch-emed/StructureDefinition/ch-emed-document-pharmaceuticaladvice|5.0.0-ballot)",
            "expression": [
                "Bundle"
            ]
        },
        {
            "extension": [
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-line",
                    "valueInteger": 1
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-col",
                    "valueInteger": 949
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-source",
                    "valueString": "TerminologyEngine"
                }
            ],
            "severity": "warning",
            "code": "invalid",
            "diagnostics": "Der Displayname f\\u00fcr http://loinc.org#61356-2 sollte einer von ''Pharmazeutische Empfehlung' (de-AT)' anstelle von 'Medication pharmaceutical advice.extended Document' sein.",
            "expression": [
                "Bundle.entry[0].resource/*Composition/Comp-ChangeDosage*/.type.coding[0].display"
            ]
        },
        {
            "extension": [
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-line",
                    "valueInteger": 1
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-col",
                    "valueInteger": 949
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-source",
                    "valueString": "TerminologyEngine"
                }
            ],
            "severity": "warning",
            "code": "invalid",
            "diagnostics": "Der Displayname f\\u00fcr http://loinc.org#61356-2 sollte einer von ''Pharmazeutische Empfehlung' (de-AT)' anstelle von 'Medication pharmaceutical advice.extended Document' sein.",
            "expression": [
                "Bundle.entry[0].resource/*Composition/Comp-ChangeDosage*/.type.display"
            ]
        },
        {
            "extension": [
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-line",
                    "valueInteger": 1
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-col",
                    "valueInteger": 1585
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-source",
                    "valueString": "TerminologyEngine"
                }
            ],
            "severity": "warning",
            "code": "invalid",
            "diagnostics": "Der Displayname f\\u00fcr http://loinc.org#61357-0 sollte einer von ''Medication pharmaceutical advice.brief Document' (en-US), 'Medication pharmaceutical advice.brief Document' (en-US), 'Medication pharm advice.brief Doc' (en-US), 'Breve asesoramiento farmacéutico sobre medicamentos.: Paciente:Punto temporal:Tipo:Documento:' (es-MX) or 'Medicatie farmaceutisch advies.kort [bevinding] (document)' (nl-NL)' anstelle von 'Medication pharmaceutical advice.brief Document' sein.",
            "expression": [
                "Bundle.entry[0].resource/*Composition/Comp-ChangeDosage*/.section[0].code.coding[0].display"
            ]
        },
        {
            "extension": [
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-line",
                    "valueInteger": 1
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-col",
                    "valueInteger": 949
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-source",
                    "valueString": "TerminologyEngine"
                }
            ],
            "severity": "warning",
            "code": "invalid",
            "diagnostics": "Der Displayname f\\u00fcr http://loinc.org#61356-2 sollte einer von ''Pharmazeutische Empfehlung' (de-AT)' anstelle von 'Medication pharmaceutical advice.extended Document' sein.",
            "expression": [
                "Bundle.entry[0].resource.type.coding[0].display"
            ]
        },
        {
            "extension": [
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-line",
                    "valueInteger": 1
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-col",
                    "valueInteger": 949
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-source",
                    "valueString": "TerminologyEngine"
                }
            ],
            "severity": "warning",
            "code": "invalid",
            "diagnostics": "Der Displayname f\\u00fcr http://loinc.org#61356-2 sollte einer von ''Medication pharmaceutical advice.extended Document' (en-US)' anstelle von 'Medication pharmaceutical advice.extended Document' sein.",
            "expression": [
                "Bundle.entry[0].resource.type.display"
            ]
        },
        {
            "extension": [
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-line",
                    "valueInteger": 1
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-col",
                    "valueInteger": 949
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-source",
                    "valueString": "TerminologyEngine"
                }
            ],
            "severity": "error",
            "code": "code-invalid",
            "diagnostics": "Der angegebene Code 'http://loinc.org#61356-2 ('Medication pharmaceutical advice.extended Document')' befindet sich nicht im ValueSet 'http://fhir.ch/ig/ch-term/ValueSet/DocumentEntry.typeCode|3.0.0'; Der Displayname f\\u00fcr http://loinc.org#61356-2 sollte einer von ''Medication pharmaceutical advice.extended Document' (en-US)' anstelle von 'Medication pharmaceutical advice.extended Document' sein.",
            "expression": [
                "Bundle.entry[0].resource.type"
            ]
        },
        {
            "extension": [
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-line",
                    "valueInteger": 1
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-col",
                    "valueInteger": 1585
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-source",
                    "valueString": "TerminologyEngine"
                }
            ],
            "severity": "warning",
            "code": "invalid",
            "diagnostics": "Der Displayname f\\u00fcr http://loinc.org#61357-0 sollte einer von ''Medication pharmaceutical advice.brief Document' (en-US), 'Medication pharmaceutical advice.brief Document' (en-US), 'Medication pharm advice.brief Doc' (en-US), 'Breve asesoramiento farmacéutico sobre medicamentos.: Paciente:Punto temporal:Tipo:Documento:' (es-MX) or 'Medicatie farmaceutisch advies.kort [bevinding] (document)' (nl-NL)' anstelle von 'Medication pharmaceutical advice.brief Document' sein.",
            "expression": [
                "Bundle.entry[0].resource.section[0].code.coding[0].display"
            ]
        },
        {
            "extension": [
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-line",
                    "valueInteger": 1
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-col",
                    "valueInteger": 6067
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-source",
                    "valueString": "TerminologyEngine"
                }
            ],
            "severity": "information",
            "code": "not-found",
            "diagnostics": "A definition for CodeSystem 'urn:oid:2.51.1.1' could not be found, so the code cannot be validated",
            "expression": [
                "Bundle.entry[6].resource/*MedicationStatement/MedStatTriatec-ChangeDosage*/.contained[0]/*Medication/Amlodipin*/.code.coding[0]"
            ]
        },
        {
            "extension": [
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-line",
                    "valueInteger": 1
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-col",
                    "valueInteger": 6186
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-source",
                    "valueString": "TerminologyEngine"
                }
            ],
            "severity": "information",
            "code": "not-found",
            "diagnostics": "A definition for CodeSystem 'urn:oid:0.4.0.127.0.16.1.1.2.1' could not be found, so the code cannot be validated",
            "expression": [
                "Bundle.entry[6].resource/*MedicationStatement/MedStatTriatec-ChangeDosage*/.contained[0]/*Medication/Amlodipin*/.form.coding[0].system"
            ]
        },
        {
            "extension": [
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-line",
                    "valueInteger": 1
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-col",
                    "valueInteger": 7628
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-source",
                    "valueString": "TerminologyEngine"
                }
            ],
            "severity": "warning",
            "code": "not-found",
            "diagnostics": "A definition for CodeSystem 'urn:oid:0.4.0.127.0.16.1.1.2.1' could not be found, so the code cannot be validated",
            "expression": [
                "Bundle.entry[6].resource/*MedicationStatement/MedStatTriatec-ChangeDosage*/.dosage[0].route.coding[0].system"
            ]
        }
    ]
}
```

**Comment by oliveregger** [comment]
not reproduable anymore with 6.3.12-snaphsot or java validator 6.3.11

**Comment by oliveregger** [comment]
fixed in v3.8.6
